test/parallel/test-stream-map.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
```

Comment:
Tests needed:

 - Test the errors from passing wrong arg values.
 - Test that concurrency > 1 indeed limits to that concurrency.
 - Test that concurrency > 1 preserves order even if `fn` calls finish out of order.
 - Test the result is a readable stream (rather than just an AsyncIterable)
 - Check `AbortController` functionality and destroying the stream when the signal is aborted